src/components/SIgnUpForm.js

```diff
@@ -0,0 +1,199 @@
+import { useState, useRef, useEffect } from 'react';
+import '../styles/signUpForm.css';
+import axios from 'axios';
+import { BASE_URL } from '../App';
+import { AiOutlineCheckCircle } from 'react-icons/ai';
+import { FaTimesCircle } from 'react-icons/fa';
+
+const PASSWORD_CHECK =
+  /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%]).{8,24}$/;
+const REGISTER_URL = `${BASE_URL}/register`;
+const SIgnUpForm = ({ handleCloseModal }) => {
+  const userRef = useRef();
+  const errRef = useRef();
+  const [createUser, setCreateUser] = useState({
+    first_name: '',
+    last_name: '',
+    email: '',
+    username: '',
+    password: '',
+    password_confirmation: '',
+  });
+
+  const fieldsCheck =
```

Comment:
You can do better than `fieldsCheck` for this variable name

Reply:
ok sir